hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/DistributionSetSelectWindow.java

```diff
@@ -197,7 +200,8 @@ private void updateTargetFilterQueryDS(final Long targetFilterQueryId, final Lon
         if (dsId != null) {
             confirmWithConsequencesDialog(tfq, dsId, actionType);
         } else {
-            targetFilterQueryManagement.updateAutoAssignDS(targetFilterQueryId, null);
+            targetFilterQueryManagement.updateAutoAssignDS(
+                    entityFactory.targetFilterQuery().updateAutoAssign(targetFilterQueryId).ds(null));
             eventBus.publish(this, CustomFilterUIEvent.UPDATED_TARGET_FILTER_QUERY);
```

Comment:
just seen from the level-of abstraction this `if` is not well balanced. Suggestion:
```
    if (ds == null) {
        confirmWithConsequencesDialog(tfq, dsId, actionType);
    } else {
        removeAutoAssignedDsFromTargetFilter(targetFilterQueryId)
    }
```
I would also suggest to move this decision out of this method and one level up - I know, not your code - but in `saveOrUpdate` there is willingly passed `null` into this method where (again) must be decided what to do.